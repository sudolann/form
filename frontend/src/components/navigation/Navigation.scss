$gray: #5A5A5A;

.nav {
    height: 85px;
    width: 200px;
    white-space: nowrap;
    position: relative;
    margin: 0 auto;

}

.nav--link {
    display: inline-block;
    padding: 10px 20px;
    text-decoration: none;
    color: $gray;
    cursor: pointer;
}