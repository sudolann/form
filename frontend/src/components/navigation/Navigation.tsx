import './Navigation.scss';
import React, { FunctionComponent, ReactElement } from 'react';
import { NavLink } from 'react-router-dom';

export const Navigation: FunctionComponent = (): ReactElement => (
    <div className="nav">
        <NavLink to="/" className="nav--link">
            Events List
        </NavLink>
        <NavLink to="/form" className="nav--link">
            Add new event
        </NavLink>
    </div>
);

