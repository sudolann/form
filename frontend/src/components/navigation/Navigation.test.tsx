import React from 'react';
import { Navigation } from './Navigation';
import '@testing-library/jest-dom/extend-expect';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
describe('Navigation', () => {
  it('should take a snapshot', async () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router history={history}>
        <Navigation />
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders navigation links', async () => {
    const history = createMemoryHistory();
    const { getByText } = render(
      <Router history={history}>
        <Navigation />
      </Router>,
    );
    expect(getByText('Events list').closest('a')).toHaveAttribute(
      'href',
      '/',
    );

    expect(getByText('Add new event').closest('a')).toHaveAttribute(
      'href',
      '/form',
    );
  });
});
