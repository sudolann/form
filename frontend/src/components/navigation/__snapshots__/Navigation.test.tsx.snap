// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="nav"
  >
    <a
      aria-current="page"
      class="nav--link active"
      href="/"
    >
      Events list
    </a>
    <a
      class="nav--link"
      href="/form"
    >
      Add new event
    </a>
  </div>
</DocumentFragment>
`;
